cmd/ingest/main.go

```diff
@@ -90,107 +93,107 @@ func main() {
 		}).Fatal("couldn't create SQL tables")
 	}
 
-	// Initialize ZMQ
-	ctx, err := zmq.NewContext()
+	//Initialize RPC connection with full node zcashd
+	rpcClient, err := frontend.NewZRPCFromConf(opts.zcashConfPath)
 	if err != nil {
 		log.WithFields(logrus.Fields{
 			"error": err,
-		}).Fatal("couldn't create zmq context")
-	}
-	defer ctx.Term()
+		}).Warn("zcash.conf failed, will try empty credentials for rpc")
 
-	// WARNING: The Socket is not thread safe. This means that you cannot
-	// access the same Socket from different goroutines without using something
-	// like a mutex.
-	sock, err := ctx.NewSocket(zmq.SUB)
-	if err != nil {
-		log.WithFields(logrus.Fields{
-			"error": err,
-		}).Fatal("couldn't create zmq context socket")
+		//Default to testnet, but user MUST specify rpcuser and rpcpassword in zcash.conf; no default
+		rpcClient, err = frontend.NewZRPCFromCreds("127.0.0.1:18232", " ", " ")
```

Comment:
Shouldn't these be empty strings for name and password, not spaces?

Reply:
Yes, I wasn't sure what to fudge there since the user has to specify these in `zcash.conf` explicitly. Please let me know if there is a default work around